shipping/environment.py

```diff
@@ -238,26 +238,28 @@ def reset(self):
         self.ship_position = current_port
 
         return self._build_state()
-
+    
     def sample_action(self):
         current_port_idx = self._get_current_port_idx()
-        if self.destination_port_index is None and current_port_idx is not None:
-            # If no destination port selected and currently at port, randomly return an index
-
-            # ** Get a port that is not my current port
-            port_index = self._sample_random_port()
-            while port_index == current_port_idx: port_index = self._sample_random_port()
-
-            return [ActionType.SELECT_PORT, port_index]
-        elif self.cargo == 0 and current_port_idx is not None:
-            # If destination port selected and currently at port, but no cargo
-            return [ActionType.TAKE_CARGO, self._sample_random_cargo(current_port_idx)]
-        elif self.fuel == 0 and current_port_idx is not None:
-            # If destination port selected and currently at port, but no fuel
-            return [ActionType.TAKE_FUEL, self._sample_random_fuel(current_port_idx)]
-        else:
-            # If destination port is selected, move a random location
+        if current_port_idx is None:
```

Comment:
Is current_port_idx ever None. I was thinking of checking ship position with destination port position.

Reply:
Now that you mention it, there was a critical bug in the finding which port the ship is on